### PR TITLE
 Utopia 229 ppq

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):
ticket UTOPIA 229
Screen Shot 2022-10-12 at 10 36 56 AM

Also add husky config for both backend and frontend

please note: i am still working on css and unit tesing, this just a preview for everyone

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [x ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ x] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

### CODE REVIEW

File: src/frontend/src/pages/PPQPage.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    <span>
+                        {' '}
+                        <b>Estimated time:</b> 20 minutes
+                    </span>
+                </div>
+                <div data-cy="ppq-img" className="ppq-svg">
+                    <img src={ppqImg} alt="Fill form image" />
+                </div>
+            </section>
+            <Footer />
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PPQPage.tsx:58]: Unnecessary Footer

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/PPQPage.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                            Get started{' '}
+                            <FontAwesomeIcon
+                                className="icon"
+                                icon={faChevronRight}
+                            />
+                        </a>
+                    </div>
+                    <br />
+                    <span>
+                        {' '}
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PPQPage.tsx:50]: Is extra space needed here? 🤔 and at other places like end of a paragraphs

File: src/frontend/tsconfig.json [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+        "forceConsistentCasingInFileNames": true,
+        "module": "ESNext",
+        "moduleResolution": "Node",
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "noEmit": true,
+        "jsx": "react-jsx"
+    },
+    "include": ["src", "cypress", "cypress.config.ts"],
+    "references": [{ "path": "./tsconfig.node.json" }]
```
[the remark this answers is not in the document]
Comment (binggaofw) [reply src/frontend/tsconfig.json:20]: sure, I just copy backend prettier config to frontend and re-run lint

File: src/frontend/src/sass/_header.scss [hunk 4]
```diff
@@ -15,7 +15,7 @@ header {
   
   .logo {
     width: 200px;
-    height: 150px;
+    height: 65px;
```
[the remark this answers is not in the document]
Comment (Bottle7) [reply src/frontend/src/sass/_header.scss:18]: Yeah, that's cool. I had made it 150px in both width and height just to make it visible but that's a great catch on the vertical spacing as an issue.

[the remark this answers is not in the document]
Comment (kushal-arora-fw) [reply src/frontend/src/sass/_header.scss:18]: Amazing, this seems sorted! Great catch, Bing 👌

File: src/frontend/package.json [hunk 5]
```diff
@@ -1,68 +1,68 @@
...
+  },
+  "dependencies": {
+    "@bcgov/bc-sans": "^1.0.1",
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
+    "breakpoint-sass": "3.0.0",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-router-dom": "^6.4.2",
```
Comment (kushal-arora-fw) [inline src/frontend/package.json:27]: Hey just noticed, can we please get rid of `^` here as well?

Comment (binggaofw) [reply src/frontend/package.json:27]: sure,done

File: src/frontend/src/pages/PPQPage.tsx (hunk 1 above)
Comment (binggaofw) [reply src/frontend/src/pages/PPQPage.tsx]: the design need the footer

File: src/frontend/src/pages/PPQPage.tsx (hunk 2 above)
Comment (binggaofw) [reply src/frontend/src/pages/PPQPage.tsx]: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-child-element-spacing.md. this is the react eslint rule, please refer the link

File: .husky/pre-commit [hunk 6]
```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
-
+echo "lint code"
 cd ./src/backend
 npm run precommit
+
+cd ../frontend
+
+npm run precommit
```
Comment (akroon3r) [inline .husky/pre-commit:9]:
```suggestion
npm run precommit

```

File: src/frontend/.eslintignore [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+dist/*
+coverage/*
+# data definition files
+**/*.d.ts
+# 3rd party libs
+/src/public/
+# custom definition files
+/src/types/
+
+.eslintrc.js
```
Comment (akroon3r) [inline src/frontend/.eslintignore:13]:
```suggestion
.eslintrc.js

```

File: src/frontend/.prettierignore [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+coverage
+documentation
+logs
+node_modules
+
+# Ignore all HTML files:
+*.html
+.gitignore
+.prettierignore
+.husky
```
Comment (akroon3r) [inline src/frontend/.prettierignore:15]:
```suggestion
.husky

```

File: src/frontend/.stylelintignore [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# Third party
+**/node_modules
+
+# Build products
+.cache/
+build/
```
Comment (akroon3r) [inline src/frontend/.stylelintignore:6]:
```suggestion
build/

```

File: src/frontend/cypress/support/component-index.html [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Components App</title>
+  </head>
+  <body>
+    <div data-cy-root></div>
+  </body>
+</html>
```
Comment (akroon3r) [inline src/frontend/cypress/support/component-index.html:12]:
```suggestion
</html>

```

**Review by akroon3r — CHANGES_REQUESTED** [review]
Just some quick CLRF things that you can quickly commit my suggestions. When that's done I'll approve.

**Review by akroon3r — APPROVED** [review]

**Review by akroon3r — APPROVED** [review]

**Comment by akroon3r** [comment]
@kushal-arora-fw your changes need to be approved by yourself before we merge this.

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**